src/test/java/com/solvd/micro9/scooters/integration/kstreams/KStreamsIT.java

```diff
@@ -0,0 +1,149 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+import org.springframework.test.annotation.DirtiesContext;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@SpringBootTest(classes = {KStreamConfig.class, StreamsBuilder.class})
+@DirtiesContext
+public class KStreamsIT extends KafkaTestcontainers {
+
+    @Autowired
+    private KStreamConfig kStreamConfig;
+
+    private final Serde<RentEvent> rentEventSerde =
+            new CustomSerde<>(RentEvent.class).getSerde();
+
+    private final Serde<BigDecimal> bigDecimalSerde =
+            new CustomSerde<>(BigDecimal.class).getSerde();
+
+    private TopologyTestDriver testDriver;
+
+    private TestInputTopic<String, RentEvent> inputTopic;
+
+    private TestOutputTopic<String, BigDecimal> outputTopic;
+
+    private KeyValueStore<String, BigDecimal> usersExpensesStore;
+
+    @BeforeEach
+    void setup() {
+        StreamsBuilder streamsBuilder = new StreamsBuilder();
+        kStreamConfig.buildPipeline(streamsBuilder);
+        testDriver = new TopologyTestDriver(
+                streamsBuilder.build(), this.getKafkaStreamProperties()
+        );
+        Serializer<RentEvent> rentEventSerializer = rentEventSerde.serializer();
+        Map<String, Object> props = new HashMap<>();
+        props.put(JsonSerializer.ADD_TYPE_INFO_HEADERS, false);
+        rentEventSerializer.configure(props, false);
+        inputTopic = testDriver.createInputTopic(
+                "rent-events",
+                Serdes.String().serializer(),
+                rentEventSerializer
+        );
+        outputTopic = testDriver.createOutputTopic(
+                "income",
+                Serdes.String().deserializer(),
+                bigDecimalSerde.deserializer()
+        );
+        usersExpensesStore =
+                testDriver.getKeyValueStore(KStreamConfig.USER_EXPENSES_STORE);
+    }
+
+    @AfterEach
+    void teardown() {
+        testDriver.close();
+    }
+
+    @Test
+    void verifyInputTopicProcessedAndSentToOutputTopic() {
+        Assertions.assertTrue(outputTopic.isEmpty());
+        RentEvent rentEvent = new FkRentEvent();
+        RentEvent rentEventWithUSD = new FkRentEvent();
+        rentEventWithUSD.setCurrency(Currency.USD);
+        RentEvent rentEventEndedNotToday = new FkRentEvent();
+        rentEventEndedNotToday.setEnded(LocalDateTime.now().withYear(2020));
+        List<RentEvent> rentEvents = List.of(
+                rentEvent,
+                rentEventWithUSD,
+                rentEventEndedNotToday
+        );
+        BigDecimal resultIncome = BigDecimal.ZERO;
+        for (RentEvent event : rentEvents) {
```

Comment:
@eyakauleva 
this logic must not be there
test is single object construction and then asserting this constructed object